clang/lib/CIR/Dialect/Transforms/LifetimeCheck.cpp

```diff
@@ -1511,8 +1733,106 @@ void LifetimeCheckPass::checkMoveAssignment(CallOp callOp,
     checkNonConstUseOfOwner(dst, callOp.getLoc());
 
     // 2.4.2 - It is an error to use a moved-from object.
-    // To that intent we mark src's pset with invalid.
-    markPsetInvalid(src, InvalidStyle::MovedFrom, callOp.getLoc());
+    markOwnerAsMovedFrom(src, callOp.getLoc());
+  }
+}
+
+void LifetimeCheckPass::checkMoveViaCall(CallOp callOp) {
+  // Detect use-after-move for types passed to rvalue reference parameters
+  // Skip if no callee
+  if (!callOp.getCallee())
+    return;
+
+  auto calleeFuncOp = getCalleeFromSymbol(theModule, *callOp.getCallee());
+  if (!calleeFuncOp)
+    return;
+
+  // Get AST function declaration to check parameter types
+  auto astAttr = calleeFuncOp.getAstAttr();
+  if (!astAttr)
+    return;
+
+  auto funcDeclAttr = dyn_cast<cir::ASTFunctionDeclAttr>(astAttr);
+  if (!funcDeclAttr)
+    return;
+
+  auto *clangFuncDecl = funcDeclAttr.getAst();
+
+  // Check each argument for rvalue reference parameters
+  unsigned numArgs = callOp.getNumArgOperands();
+  for (unsigned i = 0; i < numArgs; ++i) {
+    if (i >= clangFuncDecl->getNumParams())
```

Comment:
Add another helper function that check each param, something like `checkArgForRValueRef`

Reply:
Created checkArgForRValueRef() helper method at LifetimeCheck.cpp:1659, see 53869e9c70c5